nttt/cleanup_markdown.py

```diff
@@ -1,8 +1,19 @@
 import re
+import logging
+logging.basicConfig(filename='example.log',level=logging.INFO)
 
 
 def trim_md_tags(file_content):
-    return re.sub(r'(?P<tag>`|_{1,3}|\*{1,3})(?P<content>.+?)(?P=tag)', replacement_builder, file_content)
+    myregex = r'(?P<tag>`|_{1,3}|\*{1,3})(?P<content>.+?)(?P=tag)'
+    result = re.sub(myregex, replacement_builder, file_content)
+    delta = re.findall(myregex, file_content)
+    delta2 = ['{0}{1}{0}'.format(i[0], i[1]) for i in delta]
+    delta3 = [re.sub(myregex, replacement_builder, i) for i in delta2]
+    for i, j in zip(delta2, delta3):
+        logging.info('New substitution')
+        logging.info(i)
+        logging.info(j)
+    return result
 
 
 def replacement_builder(matchobj):
```

Comment:
We could add logging directly into `replacement_builder ` because here we know the matched text and the replacement text. No need to gather replacements into an intermediate list

Reply:
I transferred the code under replacement_builder. Additionaly I substituted logging commands with file append commands. Tags are references, as well as the whole line, the string before replacement and after.

The file and the line of the file are not available in replacement_builder. If we want this feature, I should transfer these variables and make them replacement_builder arguments. Please confirm before proceeding.

Also. logging switch has to be done as well.